### DISCUSSION TITLE
"failed to write data to efivarfs: Error writing to file descriptor: Invalid argument"

### DISCUSSION THREAD

**Discussion by HughR** [fwupd/fwupd#5215] [Q&A]
I got this error message during an upgrade of "Secure Boot dbx".  It is repeatable.

What is the problem?  Is it something wrong with fwupdmgr, the upgrade itself, or my system?  What should I do?

The title printed for the update is "Upgrade UEFI dbx from 84 to 217?"
I think almost all systems got this.
On my ancient HP 700-019 `(https://support.hp.com/ca-en/document/c03795764)`
running Fedora 36, kernel-6.0.5-200.fc36.x86_64 and earlier kernels
the upgrade fails:
Perform operation? [Y|n]: y
Downloading…             [***************************************]
Decompressing…           [***************************************]
Authenticating…          [***************************************]
Waiting…                 [***************************************]
Writing…                 [***************************************]
Writing…                 [                                       ]failed to write data to efivarfs: Error writing to file descriptor: Invalid argument

**Comment by superm1** [comment]
> Release date
> 5-Jun-2013

Ancient indeed!
Your NVRAM might be filled up or the interface to do dbx updates doesn't work I would guess.  Any matching kernel logs from while you try to do this?

@hughsie my thought is specifically with dbx we should probably not keep offering the update after X number of attempts and getting error codes from the kernel.

**Reply by hughsie** [reply]
I'm tempted to add the `blocked` quirk to any of the failed reports fro the LVFS where we've had more than one of the same combination of system firmware HWIDs.

**Reply by superm1** [reply]
> I'm tempted to add the `blocked` quirk to any of the failed reports fro the LVFS where we've had more than one of the same combination of system firmware HWIDs.

Makes sense to me

**Reply by hughsie** [reply]
https://github.com/fwupd/fwupd/pull/5220 -- for my future self the export query on the LVFS was `Plugin=uefi_dbx,HostProduct,HostVendor,HostFamily,HostSku,UpdateError,DistroId,DistroVersion`

**Reply by HughR** [reply]
Thanks for everyone's help!

I suspect that you are right -- that the NVRAM is full.
(I guess that there is a chance that the RTC battery is dead, but I haven't seen any of the normal symptoms.)
Any suggestions on how to make space in the NVRAM?

Some reports (on Lenovo machines) mention removing dump* entries in efivars but my system has none.

Here are the top entries from "ls -lS /sys/firmware/efi/efivars"
-rw-r--r--. 1 root root 4604 Oct 31 07:59 db-d719b2cb-3d3a-4596-a3bc-dad00e67656f
-rw-r--r--. 1 root root 4120 Oct 31 07:59 dbx-d719b2cb-3d3a-4596-a3bc-dad00e67656f
-rw-r--r--. 1 root root 4068 Oct 31 07:59 MrcS3Resume-87f22dcb-7304-4105-bb7c-317143ccc23b
-rw-r--r--. 1 root root 3097 Oct 31 07:59 StdDefaults-4599d26f-1a11-49b8-b91f-858745cff824
-rw-r--r--. 1 root root 1564 Oct 31 07:59 KEK-8be4df61-93ca-11d2-aa0d-00e098032b8c
-rw-r--r--. 1 root root 1467 Oct 31 07:59 PK-8be4df61-93ca-11d2-aa0d-00e098032b8c
-rw-r--r--. 1 root root 1284 Oct 31 07:59 DmiArray-4b3082a3-80c6-4d7e-9cd0-583917265df1
-rw-r--r--. 1 root root 1183 Oct 31 07:59 Setup-ec87d643-eba4-4bb5-a1e5-3f3e36b20da9
-rw-r--r--. 1 root root 1002 Oct 31 07:59 SystemIds-b088f8c5-d551-4908-98b9-d3d8b90a05ac
-rw-r--r--. 1 root root  924 Oct 31 07:59 MokListRT-605dab50-e046-4300-abb6-3dd810dd8b23
-rw-r--r--. 1 root root  389 Oct 31 07:59 Boot0008-8be4df61-93ca-11d2-aa0d-00e098032b8c

Is the large dbx-d719b2cb-3d3a-4596-a3bc-dad00e67656f a result of the fwupd that failed?  If so, how many of the other large entries are from that?  Could that stuff somehow be stored in the ESP instead?  Which of these are now useless and can be safely deleted?  I worry that a full efivars will cause other problems.

**Reply by superm1** [reply]
My proposal:
1) The easiest idea that comes to mind is you can try to do a "reset BIOS default settings".   
2) Next try to go look for a BIOS upgrade for the system.
3) If that doesn't work, with this ancient machine maybe stuff really is just broken.
